### PR TITLE
Update nginx.chart.py

### PR DESCRIPTION
**Pull request by Rex-git-dev**
Tengine version: Tengine/2.3.1
nginx version: nginx/1.16.0

stub status format 

Active connections: 3
server accepts handled requests request_time
109    109     3166    34677
Reading: 0 Writing: 1 Waiting: 2

### CODE REVIEW

**Comment by ilyam8** [comment]
Hi @Rex-git-dev 

`nginx` added `request_time` to the `stub_status` output?

I see your changes break compatibility with prev `nginx` versions.

go version handles tengine stub_status btw

 > https://github.com/netdata/go.d.plugin/tree/master/modules/nginx

And i see there is a tengine module that uses `ngx_http_reqstat_module`

> https://github.com/netdata/go.d.plugin/tree/master/modules/tengine

File: collectors/python.d.plugin/nginx/nginx.chart.py [hunk 1]
```diff
@@ -61,12 +61,12 @@ def _get_data(self):
...
-                    'waiting': int(raw[15]),
-                    'accepts': int(raw[7]),
-                    'handled': int(raw[8])}
+           return {'active': int(raw[2]),
+                    'requests': int(raw[10]),
+                    'reading': int(raw[12]),
+                    'writing': int(raw[14]),
+                    'waiting': int(raw[16]),
+                    'accepts': int(raw[8]),
+                    'handled': int(raw[9])}
```
Comment (ilyam8) [inline collectors/python.d.plugin/nginx/nginx.chart.py]: i think we need an `if` there. if `request_time` in raw or based on len(raw). To not break parsing old/not tengine stub status.

```python
if condition:
    return old
return new
```

**Comment by ilyam8** [comment]
Hi @Rex-git-dev, are you going to finish the PR?

**Comment by ilyam8** [comment]
Closing this due to inactivity

File: collectors/python.d.plugin/nginx/nginx.chart.py (hunk 1 above)
Comment (Rex-git-dev) [reply collectors/python.d.plugin/nginx/nginx.chart.py]: modify 'requests':str(raw[11]),
ValueError: invalid literal for int() with base 10: '155535228875\nReading:


mybe,status respon 
Active connections: 240 \nserver accepts handled requests request_time\n 21329772 21329772 110166559 155336225319\nReading: 0 Writing: 166 Waiting: 78

**Comment by Rex-git-dev** [comment]
raw = self._get_raw_data().replace("\n"," ").split(" ")
            return {'active': int(raw[2]),
                    'requests':int(raw[13]),
                    'reading': int(raw[15]),
                    'writing': int(raw[17]),
                    'waiting': int(raw[19]),
                    'accepts': int(raw[11]),
                    'handled': int(raw[12])}

File: collectors/python.d.plugin/nginx/nginx.chart.py (hunk 1 above)
Comment (Rex-git-dev) [reply collectors/python.d.plugin/nginx/nginx.chart.py]: raw = self._get_raw_data().replace("\n"," ").split(" ")
            return {'active': int(raw[2]),
                    'requests':int(raw[13]),
                    'reading': int(raw[15]),
                    'writing': int(raw[17]),
                    'waiting': int(raw[19]),
                    'accepts': int(raw[11]),
                    'handled': int(raw[12])}

**Comment by ilyam8** [comment]
@Rex-git-dev i dont get

nginx

```
Active connections: 1
server accepts handled requests
36 36 126
Reading: 0 Writing: 1 Waiting: 0
```

tengine

```
Active connections: 1
server accepts handled requests request_time
1140 1140 1140 75806
Reading: 0 Writing: 1 Waiting: 0
```

we need:
 1. split
 2. check if it is tengine or nginx (`request_time` in data)
 3. use appropriate indexing based on 2.

**Comment by Rex-git-dev** [comment]
> @Rex-git-dev i dont get
> 
> nginx
> 
> ```
> Active connections: 1
> server accepts handled requests
> 36 36 126
> Reading: 0 Writing: 1 Waiting: 0
> ```
> 
> tengine
> 
> ```
> Active connections: 1
> server accepts handled requests request_time
> 1140 1140 1140 75806
> Reading: 0 Writing: 1 Waiting: 0
> ```
> 
> we need:
> 
> 1. split
> 2. check if it is tengine or nginx (`request_time` in data)
> 3. use appropriate indexing based on 2.
yep

**Pull request closed without merging**